doc/writing-python-agents.md

```diff
@@ -0,0 +1,90 @@
+# Resource Agent guide for Python
```

Comment:
@krig I think we should move this under dev-guides.

Reply:
Sure, I'll make a PR.